File: backend/Ricettario.API/Models/Recipe.cs
namespace Ricettario.API.Models;

public class Recipe
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Instructions { get; set; }
    public double PrepTimeMinutes { get; set; }
    public double CookTimeMinutes { get; set; }
    public string? Difficulty { get; set; } // Easy, Medium, Hard
    public string? ImageUrl { get; set; }
    
    // For Baker's Math
    public double TotalFlourWeight { get; set; }
    
    // Serving Info
    public int ServingPieces { get; set; }
    public double? PieceWeight { get; set; }

    // Category (Many-to-One)
    public int? CategoryId { get; set; }
    public Category? Category { get; set; }

    public string UserId { get; set; } = string.Empty;
    public ApplicationUser? User { get; set; }

    public List<Ingredient> Ingredients { get; set; } = new();
    
    // Phases & Videos
    public List<RecipePhase> Phases { get; set; } = new();
    public List<RecipeVideo> Videos { get; set; } = new();
    
    // Tags (Many-to-Many)
    public List<RecipeTag> RecipeTags { get; set; } = new();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/app/models/recipe.ts
import { Category } from './category';
import { Tag } from './tag';

export interface Ingredient {
  id?: number;
  name: string;
  quantity: number;
  unit: string;
  bakersPercentage?: number;
}

export interface PhaseIngredient {
    id?: number;
    ingredientName: string;
    quantity: number;
    unit: string;
}

export interface RecipePhase {
    id?: number;
    title: string;
    recipePhaseTypeId: number;
    recipePhaseType?: { 
        id: number; 
        name: string; 
        description?: string;
        icon?: string;
        color?: string;
        isActiveWork: boolean; 
    };
    description?: string;
    durationMinutes: number;
    temperature?: number;
    ovenMode?: string;
    phaseIngredients: PhaseIngredient[];
}

export interface RecipeVideo {
    id?: number;
    url: string;
}

export interface Recipe {
  id?: number;
  title: string;
  description?: string;
  instructions?: string;
  prepTimeMinutes: number;
  cookTimeMinutes: number;
  difficulty?: string;
  imageUrl?: string;
  totalFlourWeight: number;
  servingPieces: number;
  pieceWeight?: number;
  ingredients: Ingredient[];
  
  phases?: RecipePhase[];
  videos?: RecipeVideo[];

  // Category & Tags
  categoryId?: number | null;
  category?: Category | null;
  tags?: Tag[];
  tagIds?: number[];

  userId?: string;
  createdAt?: string;
}

